src/components/App.tsx

```diff
@@ -0,0 +1,113 @@
+import '../styles/index.css'
+import '../styles/input.css'
+import '../styles/utils.css'
+import '../styles/button.css'
+import '../styles/card.css'
+import '../styles/modal.css'
+import NumberInput from "./NumberInput.tsx"
+import Input from "./Input.tsx"
+import Card from "./Card.tsx"
+import {Dispatch, SetStateAction, useState} from "react"
+import Modal from "./Modal.tsx"
+import {CardProvider} from "../context/CardContext.tsx";
+import AddCard from "./AddCard.tsx";
+
+function App() {
+    const [cardName, setCardName] = useState('')
+    const [name, setName] = useState('')
+    const [expireDate, setExpireDate] = useState({month: '', year: ''})
+    const [cardNumber, setCardNumber] = useState({first: '', second: '', third: '', fourth: ''})
+    const [isModalOpen, setIsModalOpen] = useState(false)
+
+    const handleCardClick = () => {
+        setIsModalOpen(true)
+    }
+
+    const handleCardNameChange = (value: string) => {
+        setCardName(value)
+        setIsModalOpen(false)
+    }
+    const handleInputChange = (value: string, field: string, setState: Dispatch<SetStateAction<any>>) => {
+        setState(prevState => ({
+            ...prevState,
+            [field]: value
+        }))
+    }
+
+    const handleExpiredDateChange = (value: string, field: string) => {
+        handleInputChange(value, field, setExpireDate)
+    }
+
+    const handleCardNumberChange = (value: string, field: string) => {
+        handleInputChange(value, field, setCardNumber)
+    }
+
+    return (
+        <CardProvider>
+            <>
+                <div className="app">
+                    <h2 className={"page-title"}>&lt; 카드 추가</h2>
+                    <Card cardName={cardName} cardNumber={cardNumber} name={name} cardExpireDate={expireDate}
+                          onClick={handleCardClick}></Card>
+                    <div className="input-container">
+                        <span className="input-title">카드 번호</span>
```

Comment:
r: NumberInput 컴포넌트로 숫자입력 잘 분리해주셨네요. 나머지 컴포넌트들도 미션 요구사항인 atomic한 컴포넌트를 구현해 bottom-up 방식으로 UI를 구성하는 방식으로 변경해주시기 바랍니다!
```suggestion
                        <Label>카드 번호</Label>
```